modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPFailureTest.java

```diff
@@ -179,4 +202,244 @@ public void testPatchIdPNonExistentProperties() throws IOException {
+//                userDefinedAuthenticatorPayload.convertToJasonPayload());
+//        Response response = getResponseOfPost(IDP_API_BASE_PATH, body);
+//        response.then()
+//                .log().ifValidationFails()
+//                .assertThat()
+//                .statusCode(HttpStatus.SC_CREATED)
+//                .header(HttpHeaders.LOCATION, notNullValue());
+//
+//        String location = response.getHeader(HttpHeaders.LOCATION);
+//        assertNotNull(location);
+//        String customIdPId = location.substring(location.lastIndexOf("/") + 1);
+//        assertNotNull(customIdPId);
+//    }
+
+    @Test
+    public void testAddIdPWithUserDefinedAuthenticatorWhenAuthenticatorNameIsDuplicated() throws IOException {
+
+        userDefinedAuthenticatorPayload =
+                createUserDefinedAuthenticatorPayload(AUTHENTICATOR_ID_1,
+                        "https://abc.com/authenticate",
+                        "testUser",
+                        "testPassword");
+        Response response = createUserDefAuthenticator("CustomAuthIDP1", userDefinedAuthenticatorPayload);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+        customIdPId = location.substring(location.lastIndexOf("/") + 1);
+
+        // duplicate the authenticator creation
+        duplicatedUserDefinedAuthenticatorPayload =
+                createUserDefinedAuthenticatorPayload(AUTHENTICATOR_ID_1,
+                        "https://xyz.com/authenticate",
+                        "testUser1",
+                        "testPassword1");
+        Response responseOfDuplicate = createUserDefAuthenticator("CustomAuthIDP2",
+                duplicatedUserDefinedAuthenticatorPayload);
+        responseOfDuplicate.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
+                .body("message", equalTo("Federated authenticator name " +
+                        new String(Base64.getDecoder().decode(AUTHENTICATOR_ID_1)) + " is already taken."));
+    }
+
+    @Test
+    public void testAddIdPWithUserDefinedAuthenticatorWithMultipleAuthenticators() throws IOException {
+
+        String idpCreateErrorPayload = readResource("add-idp-with-custom-fed-multi-auth.json");
+
+        UserDefinedAuthenticatorPayload userDefinedAuthenticatorPayload1 = createUserDefinedAuthenticatorPayload(
+                "Y3VzdG9tQXV0aGVudGljYXRvclg=",
+                "https://abc.com/authenticate",
+                "testUser",
+                "testPassword");
+        UserDefinedAuthenticatorPayload userDefinedAuthenticatorPayload2 = createUserDefinedAuthenticatorPayload(
+                "Y3VzdG9tQXV0aGVudGljYXRvclg=",
```

Comment:
Use deferent authenticator ids

Reply:
This is updated with 204aaea3e005918dd0879436e47ffd04e39ad2a8